src/devtools/client/debugger/src/components/Editor/EmptyLines.tsx

```diff
@@ -1,109 +1,92 @@
 
-  componentDidUpdate() {
-    this.disableEmptyLinesRaf();
-  }
+      // Attempt to update just the markers for just the 100-line blocks  surrounding
+      // the current line number
+      editor.codeMirror.operation(() => {
+        for (let hitCountChunk of uniqueChunks) {
+          const { lower, upper } = hitCountChunk;
 
-  componentWillUnmount() {
-    if (this._animationFrameId) {
-      cancelAnimationFrame(this._animationFrameId);
+          if (drawnRanges.has(lower)) {
+            continue;
+          }
+          drawnRanges.add(lower);
+
+          editor.codeMirror.eachLine(lower, upper, lineHandle => {
+            const line = fromEditorLine(editor.codeMirror.getLineNumber(lineHandle)!);
+
+            if (breakableLines?.includes(line)) {
+              editor.codeMirror.removeLineClass(lineHandle, "line", "empty-line");
+            } else {
+              editor.codeMirror.addLineClass(lineHandle, "line", "empty-line");
+            }
+          });
+        }
+      });
     }
 
-    const { editor, lower, upper } = this.props;
-    editor.editor.off("scroll", this.disableEmptyLinesRaf);
+    editor.editor.on("scroll", disableEmptyLinesRaf);
+    editor.editor.on("change", disableEmptyLinesRaf);
+    editor.editor.on("swapDoc", disableEmptyLinesRaf);
 
-    editor.codeMirror.operation(() => {
-      editor.codeMirror.eachLine(lower, upper, (lineHandle: any) => {
-        editor.codeMirror.removeLineClass(lineHandle, "line", "empty-line");
-      });
-    });
-  }
+    disableEmptyLinesRaf();
 
-  disableEmptyLinesRaf = () => {
-    if (this._animationFrameId) {
-      cancelAnimationFrame(this._animationFrameId);
-    }
-
-    this._animationFrameId = requestAnimationFrame(() => {
-      this._animationFrameId = null;
-      this.disableEmptyLines();
-    });
-  };
-
-  disableEmptyLines() {
-    const { breakableLines, editor, lower, upper } = this.props;
-
-    // Labeled "hit counts", but it's really just breaking lines into into 100-line chunks
-    const uniqueChunks = calculateHitCountChunksForVisibleLines(editor);
-
-    // Attempt to update just the markers for just the 100-line blocks  surrounding
-    // the current line number
-    editor.codeMirror.operation(() => {
-      for (let hitCountChunk of uniqueChunks) {
-        const { lower, upper } = hitCountChunk;
-        editor.codeMirror.eachLine(lower, upper, (lineHandle: any) => {
-          const line = fromEditorLine(editor.codeMirror.getLineNumber(lineHandle)!);
-
-          if (breakableLines?.includes(line)) {
-            editor.codeMirror.removeLineClass(lineHandle, "line", "empty-line");
-          } else {
-            editor.codeMirror.addLineClass(lineHandle, "line", "empty-line");
-          }
-        });
+    return () => {
+      if (animationFrameId.current) {
+        // We _want_ to clean this up if it exists
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I don't think this eslint comment will be necessary either if we use a local variable.